21-Hello-World/01-FP-Philosophical-Foundations/05-Type-Classes.md

```diff
@@ -21,23 +21,92 @@ functionName Apple = "Apple"
 
 Putting it differently, if `Some type` can implement some `function(s)/value(s) with a specified type signature` in such a way that the implementation adheres to `specific laws`, one can say it **has** an instance of `some CT concept/term`. Some types cannot satisfy some CT terms' conditions, others can satisfy them in only one way; and still others can satisfy them in multiple ways. Thus, one does not say "`Type X` **is** an instance of [some CT Term]." Rather, one says "`Type X` **has** an instance of [some CT Term]." To see this concept in a clearer way and using pictures, see https://www.youtube.com/watch?v=iJ7V1KXJpsE
 
-Thus, type classes abstract general concepts into an "interface" that can be implemented by various data types. They are usually an encapsulation of 1-4 things:
+Thus, type classes abstract general concepts into an "interface" that can be implemented by various data types. They are usually an encapsulation of 2-3 things:
 
-1. (Almost Always) The definition of type signatures for a single/multiple functions/values.
+1. (Always) The definition of type signatures for a single/multiple functions/values.
```

Comment:
Actually `(Almost Always)` was correct, because there are derived classes that don't themselves give any definition of functions/values, or there's just no functions/values to begin with. For example `class Partial` which doesn't have a body, remember also `class IsYesNoKind b <= YesNoKindConstraint b` from `Reification`

Reply:
True. My bias is showing through in the 'lawless typeclasses' debate, haha!

Also, technically type-level functions are lawless typeclasses, but I'd say those type classes aren't actual type classes...